db/view/view_update_generator.cc

```diff
@@ -88,7 +107,13 @@ future<> view_update_generator::register_staging_sstable(sstables::shared_sstabl
     if (_as.abort_requested()) {
         return make_ready_future<>();
     }
-    _sstables_with_tables.emplace_back(std::move(sst), std::move(table));
+    auto it = _sstables_with_tables.find(table);
+    if (it == _sstables_with_tables.end()) {
+        auto ret = _sstables_with_tables.emplace(table, std::vector<sstables::shared_sstable>());
+        it = ret.first;
+    }
```

Comment:
Much simpler: `_sstables_with_tables[table].push_back(std::move(sst))`

Reply:
That's just simpler to write, indeed, but I don't like using it because the `[]` STL operator require copy constructors and as far as I remember (but maybe I am wrong here), will internally throw if the element present. In this case I think we have all constructors we need, so if I am wrong about the second reason I can switch.

Reply:
It requires a default constructor, which `std::vector<>` most certainly has. And I don't see why it would need to throw. I think we can trust the STL to have reasonable implementation of methods, instead of open-coding them.

Reply:
I recall (but maybe I am wrong) @avikivity favoring the style I just used for the exception reason.
It could be that this was true but is no longer, it could also be that I am totally misremembering it.

If all are fine with it I will switch to the `[]` implementation. In fact I agree it's more elegant.

Reply:
`operator[]` will simply return a reference to the existing element if the key is present, and create an empty element if it's not. `unordered_map::at()` is a similar function which throws if it doesn't find the key - but we don't want any throwing, so `operator[]` is the way to go.